tutorbackup/templates/backup/build/backup/Dockerfile

```
@@ -1,4 +1,10 @@
-FROM docker.io/ubuntu:22.04
+{% if DOCKER_REGISTRY_PROXY is defined %}
+ARG BASE_IMAGE={{ DOCKER_REGISTRY_PROXY }}/ubuntu:22.04
+{% else %}
+ARG BASE_IMAGE=docker.io/ubuntu:22.04
+{% endif %}
+
+FROM ${BASE_IMAGE}
```

Actually, this looks a bit too complicated to me. I'd suggest this instead:

```suggestion
FROM {{ BACKUP_BASE_IMAGE }}
```

... and then a corresponding change in `plugin.py`. If someone wanted to pull their Ubuntu image from a different source, they could just override `BACKUP_BASE_IMAGE`. Wouldn't that simplify things?

---

Sure, that would be better from one plugin perspective. My thought was that if you want to proxy the base images for x number of plugins, it would be better to add a shared variable to the deployment instead of x number of per plugin variables. 

Additionally, if we allow to override the base image version, wouldn't that suggest that this plugin should work as expected with any version? While really we are bumping the base image versions based on the supported versions in the upstream releases of Tutor and Open edX and tagging the plugin accordingly.

But, I might very well be overthinking it :) Should I go ahead and apply your suggested approach to all plugins?

---

> Sure, that would be better from one plugin perspective. My thought was that if you want to proxy the base images for x number of plugins, it would be better to add a shared variable to the deployment instead of x number of per plugin variables.

You're not wrong, but in that case that would be a variable that shouldn't be added at the plugin level — rather, it should be added upstream in Tutor, and then consumed by downstream repos. You can certainly try to submit such a patch. In the past Régis hasn't been particularly inclined towards accepting such changes, but he may have modified his stance since.

> Additionally, if we allow to override the base image version, wouldn't that suggest that this plugin should work as expected with any version?

No, I'd argue that it wouldn't. The general expectation is that the plugin works with its defaults applied. If a user overrides a variable for which we supply a default, and it breaks, it's for them to fix that.

Compare this: if a user were to set `BACKUP_K8S_CRONJOB_BACKUP_SCHEDULE` to an invalid cron expression, or `BACKUP_K8S_CRONJOB_CONCURRENCYPOLICY` to a keyword that Kubernetes does not understand, then that would be an incorrect configuration, but not one that you, the plugin author, are responsible for. If you provided *defaults* for these variables that can't work, then you would be.

---

Ack, fair enough, thank! I will go ahead and make the suggested changes.